WG_GO_CORE.md

```diff
@@ -0,0 +1,177 @@
+* `M (P1)`: There is a way to prioritize different peers to optimize network utilization
+   * Bitswap peer prioritization supports pinning local / responsive peers 
+
+
+### 🗂 Large Files
+
+Go-IPFS is fast, efficient and stable when handling and transferring large amounts of data. (“large files” = 10GB - 1TB, “many small files” = 10K-1M 1MB files). The goals in this section are exclusive of finding the data (DHT/providers etc.). This is just the speed of transferring data when already connected to IPFS nodes that have that data.
+
+* `M (P0)`: Awesome go-ipfs benchmark test suite exists comparing ipfs performance and transfer size relative to bittorrent, rsync, http, cp, dd
+* `M (P0)`: Go-IPFS bitswap (or equivalent) can transfer large files from many peers at 0.8 times the speed of BitTorrent. 
+   * Bitswap session improvements that improve transfer efficiency by reducing duplicate blocks and efficiently fetching related data from many peers simultaneously. 
+   * Graphsync selector implementation that can understand non-overlapping subsets of data to select from multiple hosts and the driver that can build up graphsync requests for many peers to maximize efficiency and throughput. 
+* `M (P0)`: Go-IPFS can transfer many small files at 0.8 times the speed of rsync (for single peers) or BitTorrent (for many peers)
+   * Bitswap session improvements
+   * Graph Exchange that can route requests efficiently
+   * Graphsync
+   * Flexible provider strategies
+* `M (P1)`: Go-IPFS can transfer large files from a single peer at 0.8 times the speed of HTTP. 
+   * Flexible provider strategies to use the DHT efficiently (and therefore not slow IPFS down while adding/fetching files) while still preserving the reachability of data. 
+* `M (P1)`: Go-IPFS can add a directory of many small files at 50% the speed of a simple local hard drive copy (or dd) (10k x 1MB files)
+   * Flexible provider strategies to limit the number of DHT provider advertisements we make
+   * Faster datastore (Badger or equivalent) that doesn’t slow down as the number of blocks stored grows large
+   * GC improvements (transactions, speed)
+* `M (P2)`: Data transferred over the network is no more than 1.2x the size of the data being transferred 
+   * Bitswap session improvements to reduce duplicates
+   * Graphsync
+
+
+### 🔄 Decentralized Web
+
+go-ipfs is the go-to backend for building DApps.
+
+* `M (P0)`: Provider lookups are efficient (<1 sec) in real-world network conditions (ex between physically distant peers)
+* `M (P0)`: ipfs://, ipns:// work in web browsers
+   * Base32 CIDs by default
+   * Base32 IPNS keys (or make IPNS keys CIDs)
+* `M (PX)`: <1 sec IPNS resolution for any IPNS record
+   * QUIC
+   * NAT detection/traversal
+   * Only "reachable" nodes join the DHT
+* `M (P1)`: IPFS provides a clean way to persist data (dapps need to store stuff)
+* `M (PX)`: IPFS supports basic dapp requirements for security and private content

```

Comment:
I don't think we're going to get to this until Q2 at the earliest, but it'd be nice to take a pass on this work before doing a security audit to make that analysis more valuable. So maybe API security in Q2 and examples/documentation in Q3?
```suggestion
* `M (PX)`: IPFS supports basic dapp requirements for security and private content
   * Q2 - dapps are secure (api security, multi tenancy). Apps can't read each other's state
   * Q3 - There are examples and documentation for ipfs users to create private-content dapps using IPFS (aka handling encryption on the client side)
```

Reply:
I agree.